pkg/sriov/sriov.go

```diff
@@ -67,108 +68,134 @@ func NewSriovManager() Manager {
+		}
+		defer func() {
+			if err != nil {
+				logging.Debug("Move the interface back to initNS because of ", "error", err)
+				initns, _ := ns.GetCurrentNS()
+				linkSetNsFdError := netlink.LinkSetNsFd(linkObj, int(initns.Fd()))
+				logging.Debug("LinkSetNsFd failed in defer", "error", linkSetNsFdError)
 			}
+		}()
+		// Rename the interface to pod interface name
+		if err = s.nLink.LinkSetName(tempNSLinkObj, podifName); err != nil {
+			return fmt.Errorf("failed to rename host device %q to %q: %v", linkName, podifName, err)
 		}
-	}
-
-	// 4. Change netns
-	logging.Debug("4. Change netns",
-		"func", "SetupVF",
-		"linkObj", linkObj,
-		"netns.Fd()", int(netns.Fd()))
-	if err := s.nLink.LinkSetNsFd(linkObj, int(netns.Fd())); err != nil {
-		return fmt.Errorf("failed to move IF %s to netns: %q", tempName, err)
-	}
-
-	if err := netns.Do(func(_ ns.NetNS) error {
-		// 5. Set Pod IF name
-		logging.Debug("5. Set Pod IF name",
+		// Restore the original link name in case of error
+		defer func() {
+			if err != nil && tempNSLinkObj != nil {
+				logging.Debug("Restore the original link name because of ", "error", err)
+				linkSetNameError := s.nLink.LinkSetName(tempNSLinkObj, linkName)
+				logging.Debug("LinkSetName failed  in defer", "error", linkSetNameError)
+			}
+		}()
+		// 3. Remove alt name from the nic
+		logging.Debug("3. Remove interface original name from alt names",
 			"func", "SetupVF",
-			"linkObj", linkObj,
-			"podifName", podifName)
-		if err := s.nLink.LinkSetName(linkObj, podifName); err != nil {
-			return fmt.Errorf("error setting container interface name %s for %s", linkName, tempName)
+			"tempNSObj", tempNSLinkObj,
+			"OriginalLinkName", linkName)
+		for _, altName := range tempNSLinkObj.Attrs().AltNames {
+			if altName == linkName {
+				if err = s.nLink.LinkDelAltName(tempNSLinkObj, linkName); err != nil {
+					return fmt.Errorf("error removing VF altname %s: %v", linkName, err)
+				}
+			}
 		}
 
-		// 6. Enable IPv4 ARP notify and IPv6 Network Discovery notify
-		// Error is ignored here because enabling this feature is only a performance enhancement.
-		logging.Debug("6. Enable IPv4 ARP notify and IPv6 Network Discovery notify",
+		// 4. Change netns
+		logging.Debug("4. Change netns",
 			"func", "SetupVF",
-			"podifName", podifName)
-		_ = s.utils.EnableArpAndNdiscNotify(podifName)
+			"tempNSObj", tempNSLinkObj,
+			"netns.Fd()", int(netns.Fd()))
+		if err = s.nLink.LinkSetNsFd(tempNSLinkObj, int(netns.Fd())); err != nil {
+			return fmt.Errorf("failed to move IF %s to netns: %q", podifName, err)
+		}
 
-		// 7. Set MAC address
-		if conf.MAC != "" {
-			logging.Debug("7. Set MAC address",
+		if err = netns.Do(func(_ ns.NetNS) error {
+			// Restore the original link name in case of error
+			defer func() {
+				if err != nil {
+					logging.Debug("Move the interface back to initNS because of ", "error", err)
+					initns, _ := ns.GetCurrentNS()
```

Comment:
this returns the current namespace which is the pod sandbox ns in this case AFAIK